src/compile/data/formatparse.ts

```diff
@@ -122,14 +122,20 @@ export class ParseNode extends DataFlowNode {
         if (isTimeFieldDef(fieldDef)) {
           implicit[fieldDef.field] = 'date';
         } else if (isNumberFieldDef(fieldDef)) {
-          if (isCountingAggregateOp(fieldDef.aggregate)) {
-            return;
+          if (!isCountingAggregateOp(fieldDef.aggregate)) {
+            implicit[fieldDef.field] = 'number';
```

Comment:
1) Why is this related to sort?

2) What about distinct and some other operations that do not require numeric input?

Reply:
I wanted to make it consistent with the code below and also I don't like to use return when we might switch to a proper for loop in the future. It's a disaster bound to happen.

Re distinct. Yes, that's incorrect but unrelated to this PR.